### PR TITLE
Transmitter status home

### PR DESCRIPTION
**Pull request by AdrianLxM**
- Added Transmitter Battery Status to Homescreen
- It will be in the background - chart is drawn on top (-> should not affect visibility much)
- Sensor will now have lowest transmitter battery level reported so far for this sensor period.
- Moved constants for thresholds to class Constants.java (avoid redundancy/inconsistency)

This is how it looks like:

low
empty

Should we add "experimental interpretation" to the "rather low"-status like in the SystemStatus?
Feedback highly appreciated! :)

### CODE REVIEW

**Comment by ghost** [comment]
Good idea, I would prefer an alarm/notification in terms of transmitter battery status.

Cheers,
Matthias

**Comment by AdrianLxM** [comment]
@mgranberry @StephenBlackWasAlreadyTaken: Changed status message to "low" and "VERY LOW".

@ceben80, At least the "low" status should be shown when there is still about one month of battery. This is why we agreed not to have an alarm for that - that could ring during the night or at a meeting, when it is not actually urgent to react.
An alarm for the "Very low"/empty state would be a different pr. The future will tell if it is even necessary as users are hopefully advised early enough now.

**Comment by AdrianLxM** [comment]
Tested when setting new sensor and the battery_level gets updated correctly.

**Comment by StephenBlackWasAlreadyTaken** [comment]
This looks good to me, do we plan on ever using the background to display other alerts?

**Comment by AdrianLxM** [comment]
At the moment I cannot think of other not rapidly changing notifications that should be shown but are not urgent enough to rise a Notification/Alarm.

**Comment by jstevensog** [comment]
Would be good if these messages also appeared in widget and permanent notification, so they are always visible.
Just a thought.
Cheers

---- AdrianLxM wrote ----

**Comment by StephenBlackWasAlreadyTaken** [comment]
Merge at will!

**Comment by StephenBlackWasAlreadyTaken** [comment]
(sorry it took me a while to get around to changing my wixel to send low values for testing)

File: app/src/main/java/com/eveningoutpost/dexdrip/Home.java [hunk 1]
```diff
@@ -149,6 +154,40 @@ public void setupCharts() {
 
         chart.setZoomType(ZoomType.HORIZONTAL);
 
+        //Transmitter Battery Level
+        final Sensor sensor = Sensor.currentSensor();
+        if (sensor != null && sensor.latest_battery_level != 0 && sensor.latest_battery_level <= Constants.TRANSMITTER_BATTERY_LOW) {
```
Comment (StephenBlackWasAlreadyTaken) [inline app/src/main/java/com/eveningoutpost/dexdrip/Home.java:159]: actually should we put in a check here to never show anything if they are using dexcom share

**Comment by AdrianLxM** [comment]
@StephenBlackWasAlreadyTaken, I thought, that with checking for null and if the battery level is not 0, not showing anything for share users is done automatically.
I cannot test this though.

Did you take back the "Merge at will!"?

**Comment by StephenBlackWasAlreadyTaken** [comment]
I suppose just checking that its not zero should be fine!

**Comment by AdrianLxM** [comment]
I totally missed your last comment... merged

**Pull request merged**